platform/ui/src/components/HeaderPatientInfo/HeaderPatientInfo.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Icon } from '@ohif/ui';
+
+import OHIF, { utils } from '@ohif/core';
+
+const { formatDate } = utils;
+
+function HeaderPatientInfo({ servicesManager }) {
+  const [expanded, setExpanded] = useState(false);
+  const [patientInfo, setPatientInfo] = useState({
+    PatientName: '',
+    PatientID: '',
+    PatientSex: '',
+    PatientAge: '',
+    PatientDOB: '',
+  });
+
+  const { viewportGridService, displaySetService, cornerstoneViewportService } =
+    servicesManager.services;
+
+  const updatePatientInfo = () => {
+    const { activeViewportId, viewports } = viewportGridService.getState();
+    const activeViewport = viewports.get(activeViewportId);
+    const displaySetInstanceUID = activeViewport?.displaySetInstanceUIDs[0];
+    if (!displaySetInstanceUID) {
+      return;
+    }
+    const displaySet = displaySetService.getDisplaySetByUID(displaySetInstanceUID);
+    const image0 = displaySet.images[0];
```

Comment:
not all displaySets have images, try with dicom pdf, it will fail most likely or a video (OT) one

Reply:
Hi all, I think it would be better to check that all display sets beings rendered belongs to the same patient, and warn it if not.

The common and dangerous risk is to mix patient during image comparison.
So if display sets does not all belongs to a same patient, better to not display any name (or better a "mixed patient" warning).

If you display the name of the first display set, a physician can wrongly believe are belonging to a single patient

Reply:
It would require to check all display set of all viewport rather than taking the active viewport (as the toolbar is common to all viewports)

Reply:
I agree, we can have a fallback in that case to show the patientId on the viewport as an overlay with two different colors

Reply:
At least we should show warning, maybe we implement overlay later

Reply:
Just implemented this, you can test it locally

http://localhost:3000/viewer?StudyInstanceUIDs=1.3.6.1.4.1.14519.5.2.1.3023.4024.215308722288168917637555384485&StudyInstanceUIDs=1.3.6.1.4.1.25403.345050719074.3824.20170125095438.5

do a 3x3 grid, warning should show up, if you drop a series from the same study into all the viewports it will go away.

I also fixed the issue for OT and other modalities with no images.

If we decide to change our mind about this, there's code to also change the patient name based on the active viewport id.